src/controller/CHIPDeviceController.cpp

```diff
@@ -845,7 +845,17 @@ CHIP_ERROR DeviceCommissioner::EstablishPASEConnection(NodeId remoteDeviceId, Re
 exit:
     if (err != CHIP_NO_ERROR)
     {
-        RendezvousCleanup(err);
+        // Delete the current rendezvous session only if a device is not currently being paired.
+        if (mDeviceBeingCommissioned == nullptr)
+        {
+            FreeRendezvousSession();
+        }
```

Comment:
> Is this PR just a revert?
Yes. I didn't understand it earlier and removed the code that seemed to me to not have sense. Beside of that I added setting `mDeviceBeingCommissioned` to nullptr, as in some error cases it wasn't cleared and because of that it was not possible to perform more than one commissioning attempt.

I don't have deep knowledge regarding this code, as I only tried to fix bug I observed. Maybe @pan-apple or @bzbarsky-apple could say a little bit more on that.

Reply:
Then given that this is just a revert, and based upon @pan-apple 's confirmation that the delta here does include the necessary fix (calling ReleaseCommissioneeDevice when appropriate), this seems fine.

Cleanup of FreeRendezvousSession is probably better handled as part of #12821.